src/main/deltachat/index.js

```diff
@@ -104,84 +65,54 @@ class DeltaChatController extends EventEmitter {
     return app.translate(txt)
   }
 
-  /**
-   *
-   * @param {int} chatId
-   * @param {int} msgId
-   * @param {string} eventType
-   */
-  onMessageUpdate (chatId, msgId, eventType) {
-    this.sendToRenderer('DD_EVENT_MSG_UPDATE', { chatId, messageObj: this.messageIdToJson(msgId), eventType })
-  }
-
   checkPassword (password) {
     return password === this.getConfig('mail_pw')
   }
 
   registerEventHandler (dc) {
-    dc.on('ALL', (event, data1, data2) => {
-      this.logCoreEvent(event, data1, data2)
-    })
-
-    dc.on('DC_EVENT_CONFIGURE_PROGRESS', progress => {
-      if (Number(progress) === 0) { // login failed
-        this.emit('DC_EVENT_LOGIN_FAILED')
-        this.logout()
+    dc.on('ALL', (event, ...args) => {
+      if (!isNaN(event)) {
+        event = eventStrings[event]
       }
+      this.logCoreEvent(event, ...args)
+      if (event === 'DC_EVENT_INFO') return
+      this.sendToRenderer(event, ...args)
     })
 
-    dc.on('DC_EVENT_IMEX_FILE_WRITTEN', (filename) => {
-      this.emit('DC_EVENT_IMEX_FILE_WRITTEN', filename)
-    })
-
-    dc.on('DC_EVENT_IMEX_PROGRESS', (progress) => {
-      this.emit('DC_EVENT_IMEX_PROGRESS', progress)
-    })
-
-    dc.on('DC_EVENT_CONFIGURE_PROGRESS', progress => {
-      this.sendToRenderer('DC_EVENT_CONFIGURE_PROGRESS', progress)
-    })
-
-    dc.on('DC_EVENT_CONTACTS_CHANGED', (contactId) => {
-      this.updateChatList()
-      this.updateBlockedContacts()
-    })
-
-    dc.on('DC_EVENT_CHAT_MODIFIED', (chatId) => {
-      log.debug('DC_EVENT_CHAT_MODIFIED: ' + chatId)
-      this.chatModified(chatId)
-    })
+    dc.on('DD_EVENT_CHATLIST_UPDATED', this.onChatListChanged.bind(this))
 
     dc.on('DC_EVENT_MSGS_CHANGED', (chatId, msgId) => {
-      if (chatId === 0) {
-        this.updateChatList()
-        return
-      }
-      // Don't update if a draft changes
-      if (msgId === 0) return
+      this.onChatListChanged()
+      this.onChatListItemChanged(chatId)
       this.onMessageUpdate(chatId, msgId, 'DC_EVENT_MSGS_CHANGED')
```

Comment:
@nicodh the reason why we call onChatListChanged and onChatListItemChanged is because onChatListChanged triggers a refetch of the chatlist, which would result in a resorting of the displayed chatlists. The useLazyChatListItems hook only tries to fetch chats in view after chatListIds changed, but doesn't force refetch them (this is needed for search, where only the sort/which ids are shown changes but we don't want to refetch all chats). And this is where onChatListItemChanged comes into play, because if called with chatId=0, it force refetches all chats in view. It probably makes sense to split this into two events, onChatListItemChanged onChatListForceUpdateInView or something like this.